Library/Homebrew/cmd/update.sh

```diff
@@ -640,10 +631,21 @@ EOS
 
       # HOMEBREW_UPDATE_FORCE and HOMEBREW_UPDATE_AUTO aren't modified here so ignore subshell warning.
       # shellcheck disable=SC2030
-      if [[ "${UPSTREAM_REPOSITORY_URL}" == "https://github.com/"* ]]
+      if [[ "${UPSTREAM_REPOSITORY_URL}" =~ https://(([^:@]+)(:([^@]+))?@)?github.com/(.*)$ ]]
```

Comment:
```suggestion
      if [[ "${UPSTREAM_REPOSITORY_URL}" == "https://github.com/"* ]] || 
         [[ "${UPSTREAM_REPOSITORY_URL}" =~ https://(([^:@]+)(:([^@]+))?@)?github.com/(.*)$ ]]
```

This may allow the regex to be made simpler and ensures the previous logic takes priority.

Reply:
@sazriel26 This change didn't seem to be made. If there's a token in the URL: it will never match `"https://github.com/"*` so that conditional should always be present first.